ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -328,6 +329,14 @@ forkBlockProduction maxBlockBodySize IS{..} BlockProduction{..} =
             case mIsLeader of
               Nothing    -> return NotLeader
               Just proof -> do
+                -- We need to measure the trace just before we actually run the
+                -- protocol because it is possible that the the __current node
+                -- is not the slot leader__ and will __not lead__ the protocol/block
+                -- production.
+                --
+                -- TODO(KS): We need to add the mempool size for measurement.
```

Comment:
No need for this TODO, we have a ticket open for it (#1431) that will shortly be addressed.